src/main/java/org/jenkins/plugins/lockableresources/LockStep.java

```diff
@@ -0,0 +1,65 @@
+package org.jenkins.plugins.lockableresources;
+
+import java.io.Serializable;
+
+import org.jenkinsci.plugins.workflow.steps.AbstractStepDescriptorImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+
+import hudson.Extension;
+import hudson.model.AutoCompletionCandidates;
+
+public class LockStep extends AbstractStepImpl implements Serializable {
+
+	public final String resource;
+
+	public boolean inversePrecedence = false;
+
+	@DataBoundConstructor
+	public LockStep(String resource) {
+		if (resource == null || resource.isEmpty()) {
+			throw new IllegalArgumentException("must specify resource");
+		}
+		if (LockableResourcesManager.get().fromName(resource) == null) {
+			throw new IllegalArgumentException("resource [" + resource + "] does not exist, missing global definition?");
```

Comment:
Already answered in the other comment. Creating resources as the build goes does not make much sense, as a new resource would be created on each build, unless we somehow uniquely identify a `lock` step and link it to a generated resource, which seems overkill to me.

Moreover, Freestyle interoperability would require to create resources in global configuration.

This case is similar to `tool`, which requires the tool to be defined globally first.